sdk/tables/azure-data-tables/azure/data/tables/_table_batch.py

```diff
@@ -34,54 +38,47 @@ class TableBatchOperations(object):
     entities, and its total payload may be no more than 4 MB in size.
 
     :ivar str table_name: The name of the table.
-    :ivar requests: A list of :class:`~azure.core.pipeline.transport.HttpRequest` in a batch.
-    :vartype requests: list[~azure.core.pipeline.transport.HttpRequest]
+    :ivar requests: A list of :class:`~azure.core.rest.HttpRequest` in a batch.
+    :vartype requests: list[~azure.core.rest.HttpRequest]
     """
 
     def __init__(
         self,
-        client: AzureTable,
-        serializer: Serializer,
-        deserializer: Deserializer,
-        config: AzureTableConfiguration,
+        config: Union[AzureTableConfiguration, AsyncAzureTableConfiguration],
+        endpoint: str,
         table_name: str,
         is_cosmos_endpoint: bool = False,
         **kwargs,
     ) -> None:
         """Create TableClient from a Credential.
 
-        :param client: An AzureTable object.
-        :type client: ~azure.data.tables._generated.AzureTable
-        :param serializer: A Serializer object for request serialization.
-        :type serializer: ~azure.data.tables._generated._serialization.Serializer
-        :param deserializer: A Deserializer object for request deserialization.
-        :type deserializer: ~azure.data.tables._generated._serialization.Deserializer
         :param config: An AzureTableConfiguration object.
         :type config: ~azure.data.tables._generated._configuration.AzureTableConfiguration
+        :param endpoint: The primary account URL.
+        :type endpoint: str
         :param table_name: The name of the Table to perform operations on.
         :type table_name: str
         :param is_cosmos_endpoint: True if the client endpoint is for Tables Cosmos. False if not. Default is False.
         :type is_cosmos_endpoint: bool
-        :returns: None
         """
-        self._client = client
-        self._serialize = serializer
-        self._deserialize = deserializer
-        self._config = config
-        self._is_cosmos_endpoint = is_cosmos_endpoint
-        self.table_name = table_name
+        self._config: Union[AzureTableConfiguration, AsyncAzureTableConfiguration] = config
+        self._base_url: str = endpoint
+        self._is_cosmos_endpoint: bool = is_cosmos_endpoint
+        self.table_name: str = table_name
 
-        self._partition_key = kwargs.pop("partition_key", None)
-        self.requests = []  # type: List[HttpRequest]
+        self._partition_key: Optional[str] = kwargs.pop("partition_key", None)
```

Comment:
Looks we forgot `:keyword partition_key` in docstring.

Reply:
This is a good catch - I don't think this should be here at all.